File: Java/Easy/1816.java
class Solution {
    public String truncateSentence(String s, int k) {
        // count number of spaces
        // then use substring function to get the actual string
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == ' ') {
                k--;
                if (k == 0) {
                    return s.substring(0, i);
                }
            }

        }

        return s;
    }
}

File: C/Easy/1816.c
char *truncateSentence(char *s, int k)
{
    char temp;
    int count = 0;
    for (int i = 0; i < strlen(s) - 1; i++)
    {
        if (s[i] == ' ')
        {
            count++;
        }
        if (count == k)
        {
            s[i] = '\0';
        }
    }
    return s;
}